spring-integration-zip/src/test/java/org/springframework/integration/zip/transformer/UnZipTransformerTests.java

```diff
@@ -71,6 +72,37 @@ public void setup() throws IOException {
 		this.workDir = testFolder.newFolder();
 	}
 
+
+	/**
+	 * test flat file entry structure without additional dictory entries
+	 *
+	 * @throws IOException
+	 */
+	@Test
+	public void unzipFlatFileEntryZip() throws IOException {
+		File singleFile = this.resourceLoader.getResource("classpath:testzipdata/single.txt").getFile();
+		File zip = new File(singleFile.getParentFile(), "flatfileentry.zip");
```

Comment:
Well, I think we need to create a final zip file exactl in the `testFolder`, so it will be removed after we exit from tests.
In the current your solution it is always going to appear in the same directory as your original `single.txt` for zipping.

Also, please, add your name to the `@author` list for all the affected classes.

Reply:
I'm not sure if I understand you correctly. Please have a look on my next commit.